metricbeat/module/docker/diskio/helper.go

```diff
@@ -45,56 +48,62 @@ func (io *BLkioService) getBlkioStatsList(rawStats []docker.Stat, dedot bool) []
 	return formattedStats
 }
 
-func (io *BLkioService) getBlkioStats(myRawStat *docker.Stat, dedot bool) BlkioStats {
+// TODO: User StorageStats instead of BlkioStats on Windows
+func (io *BlkioService) getBlkioStats(myRawStat *docker.Stat, dedot bool) BlkioStats {
 	newBlkioStats := io.getNewStats(myRawStat.Stats.Read, myRawStat.Stats.BlkioStats.IoServicedRecursive)
-	oldBlkioStats, exist := io.BlkioSTatsPerContainer[myRawStat.Container.ID]
+	bytesBlkioStats := io.getNewStats(myRawStat.Stats.Read, myRawStat.Stats.BlkioStats.IoServiceBytesRecursive)
 
 	myBlkioStats := BlkioStats{
 		Time:      myRawStat.Stats.Read,
 		Container: docker.NewContainer(myRawStat.Container, dedot),
+
+		serviced:      newBlkioStats,
+		servicedBytes: bytesBlkioStats,
 	}
 
+	oldBlkioStats, exist := io.BlkioStatsPerContainer[myRawStat.Container.ID]
 	if exist {
 		myBlkioStats.reads = io.getReadPs(&oldBlkioStats, &newBlkioStats)
 		myBlkioStats.writes = io.getWritePs(&oldBlkioStats, &newBlkioStats)
 		myBlkioStats.totals = io.getTotalPs(&oldBlkioStats, &newBlkioStats)
 	}
 
-	io.BlkioSTatsPerContainer[myRawStat.Container.ID] = newBlkioStats
+	io.BlkioStatsPerContainer[myRawStat.Container.ID] = newBlkioStats
 
 	return myBlkioStats
 }
 
-func (io *BLkioService) getNewStats(time time.Time, blkioEntry []types.BlkioStatEntry) BlkioRaw {
+func (io *BlkioService) getNewStats(time time.Time, blkioEntry []types.BlkioStatEntry) BlkioRaw {
 	stats := BlkioRaw{
 		Time:   time,
 		reads:  0,
 		writes: 0,
 		totals: 0,
 	}
+
 	for _, myEntry := range blkioEntry {
```

Comment:
Entries are per operation type, but also per Major/Minor (per device), so they should be summed by type. We could also decide to send metrics per device. Opinions?

Reply:
These blkio entries are defined as:
```
// BlkioStatEntry is one small entity to store a piece of Blkio stats
// Not used on Windows.
type BlkioStatEntry struct {
    Major uint64 `json:"major"`
    Minor uint64 `json:"minor"`
    Op    string `json:"op"`
    Value uint64 `json:"value"`
}
```
I mean Major/Minor numbers as the identifier of a device in a posix system, in this case they identify the block device related to these stats.

So we have information per device, and we could send that, though the mapping between Major/Minor and device files, mountpoints or something useful would be tricky, and we couldn't provide the same functionality for Windows.

Reply:
Yes, in any case I'm not sure about how to do it in a useful way with proper mapping between device ids and device files or mountpoints. We can leave it till we see someone needing it.